src/Catalyst.Core.Modules.Dfs/WebApi/V0/Controllers/PubSubController.cs

```diff
@@ -0,0 +1,142 @@
+* the Free Software Foundation, either version 2 of the License, or
+* (at your option) any later version.
+*
+* Catalyst.Node is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License
+* along with Catalyst.Node. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#endregion
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Catalyst.Abstractions.Dfs;
+using Microsoft.AspNetCore.Mvc;
+
+namespace Catalyst.Core.Modules.Dfs.WebApi.V0.Controllers
+{
+    /// <summary>
+    ///     A list of topics.
+    /// </summary>
+    public class PubsubTopicsDto
+    {
+        /// <summary>
+        ///     A list of topics.
+        /// </summary>
+        public IEnumerable<string> Strings;
+    }
+
+    /// <summary>
+    ///     A list of peers.
+    /// </summary>
+    public class PubsubPeersDto
+    {
+        /// <summary>
+        ///     A list of peer IDs.
+        /// </summary>
+        public IEnumerable<string> Strings;
+    }
+
+    /// <summary>
+    ///     Publishing and subscribing to messages on a topic.
+    /// </summary>
+    public class PubSubController : IpfsController
+    {
+        /// <summary>
+        ///     Creates a new controller.
+        /// </summary>
+        public PubSubController(IDfsService dfs) : base(dfs) { }
+
+        /// <summary>
+        ///     List all the subscribed topics.
+        /// </summary>
+        [HttpGet] [HttpPost] [Route("pubsub/ls")]
+        public async Task<PubsubTopicsDto> List()
+        {
+            return new PubsubTopicsDto
+            {
+                Strings = await IpfsCore.PubSubApi.SubscribedTopicsAsync(Cancel)
+            };
+        }
+
+        /// <summary>
+        ///     List all the peers associated with the topic.
+        /// </summary>
+        /// <param name="arg">
+        ///     The topic name or null/empty for "all topics".
+        /// </param>
+        [HttpGet] [HttpPost] [Route("pubsub/peers")]
+        public async Task<PubsubPeersDto> Peers(string arg)
+        {
+            var topic = string.IsNullOrEmpty(arg) ? null : arg;
+            var peers = await IpfsCore.PubSubApi.PeersAsync(topic, Cancel);
+            return new PubsubPeersDto
+            {
+                Strings = peers.Select(p => p.Id.ToString())
+            };
+        }
+
+        /// <summary>
+        ///     Publish a message to a topic.
+        /// </summary>
+        /// <param name="arg">
+        ///     The first arg is the topic name and second is the message.
+        /// </param>
+        [HttpGet] [HttpPost] [Route("pubsub/pub")]
+        public async Task Publish(string[] arg)
+        {
+            if (arg.Length != 2)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4832783)